src/Build.UnitTests/BinlogVersionCompatibilityTests.cs

```diff
@@ -0,0 +1,372 @@
+                    
+                    // Test reading with Structured Logger if available
+                    TestBinlogPlaybackWithStructuredLogger(binlogPath);
+                    
+                    _output.WriteLine($"✓ Successfully played back {Path.GetFileName(binlogPath)}");
+                }
+                catch (Exception ex)
+                {
+                    _output.WriteLine($"✗ Failed to play back {Path.GetFileName(binlogPath)}: {ex.Message}");
+                    failedBinlogs.Add(binlogPath);
+                }
+            }
+
+            failedBinlogs.ShouldBeEmpty($"All binlogs should be playable. Failed: {string.Join(", ", failedBinlogs.Select(Path.GetFileName))}");
+        }
+
+        private void TestBinlogPlaybackWithReplayEventSource(string binlogPath)
+        {
+            var replayEventSource = new BinaryLogReplayEventSource();
+            var events = new List<BuildEventArgs>();
+            
+            replayEventSource.AnyEventRaised += (sender, e) => events.Add(e);
+            
+            replayEventSource.Replay(binlogPath);
+            
+            events.Count.ShouldBeGreaterThan(0, "Should have received build events");
+            events.ShouldContain(e => e is BuildStartedEventArgs, "Should contain BuildStartedEventArgs");
+            events.ShouldContain(e => e is BuildFinishedEventArgs, "Should contain BuildFinishedEventArgs");
+        }
+
+        private void TestBinlogPlaybackWithStructuredLogger(string binlogPath)
+        {
+            try
+            {
+                // Test that we can successfully read the binlog using the standard MSBuild binary log reader
+                var reader = new BinaryLogReplayEventSource();
+                var projects = new List<ProjectStartedEventArgs>();
+                var targets = new List<TargetStartedEventArgs>();
+                var tasks = new List<TaskStartedEventArgs>();
+                var buildFinished = false;
+
+                reader.ProjectStarted += (sender, e) => projects.Add(e);
+                reader.TargetStarted += (sender, e) => targets.Add(e);
+                reader.TaskStarted += (sender, e) => tasks.Add(e);
+                reader.BuildFinished += (sender, e) => buildFinished = true;
+
+                reader.Replay(binlogPath);
+
+                // Verify basic build structure
+                buildFinished.ShouldBeTrue("Build should have finished");
+                projects.ShouldNotBeEmpty("Should contain projects");
+                targets.ShouldNotBeEmpty("Should contain targets");
+
+                _output.WriteLine($"  - Loaded build with {projects.Count} projects, {targets.Count} targets, {tasks.Count} tasks");
+            }
+            catch (Exception ex)
+            {
+                _output.WriteLine($"  - Error during structured analysis: {ex.Message}");
+                throw;
+            }
+        }
+
+        private bool BuildProjectWithBinlog(string binlogParameters)
```

Comment:
this will always build with the latest version of binlog. It is not trivial to build with an older version, I think you might need to checkout older versions of MSBuild and build using those, for each version.

It is also not advisable to build these binlogs on every test run, build them once manually and include as resources, to avoid wasting time on every test run. These binlogs will never change.